app/tests/e2e.transformer.test.ts

```diff
@@ -0,0 +1,132 @@
+    expect.assertions(5);
+
+    // When transformers are requested
+    const response = await fetch(tansformerBaseUrl, {
+      method: "get",
+    });
+
+    const message = await response.json();
+    /**
+     * .length call does not work on imported json modules.
+     * Imported modules a key for each member of the array, and a default key.
+     * Hence counting the number of keys and then subtracting by one.
+     */
+    const expectedLength = Object.keys(transformerGetExpected).length - 1;
+
+    // Then expect all transformers to be returned
+    expect(message.length).toBe(expectedLength);
+    expect(message[0].name).toBe(transformerGetExpected[0].name);
+    expect(message[1].name).toBe(transformerGetExpected[1].name);
+    expect(message[1].source.replace(/\s/g, "")).toBe(
+      transformerGetExpected[1].source.replace(/\s/g, "")
+    );
+    expect(response.status).toBe(200);
+  });
+
+  it("should get the transformed message with transformed status code", async () => {
+    expect.assertions(2);
+
+    // When a get request is sent after creating a new transformer
+    const response = await fetch(baseUrl, {
+      method: "get",
+    });
+
+    const message = await response.json();
+
+    // Then expect the new transformer to modify the message and status code
+    expect(response.status).toBe(418);
+    expect(message).toEqual({ message: "Newly transformed" });
+  });
+
+  it("should get a transformer by id and delete it", async () => {
+    expect.assertions(4);
+
+    // Given there are existing transformers
+    const response = await fetch(tansformerBaseUrl, {
+      method: "get",
+    });
+
+    expect(response.status).toBe(200);
+
+    // Get the id of the transformer to be deleted
+    const message = await response.json();
+    const toDeleteId = message.filter(
+      (element: any) => element.name === testTransformer
+    )[0].id;
+
+    // When the transformer to be deleted is requested by id
+    const responseById = await fetch(`${tansformerBaseUrl}/${toDeleteId}`, {
+      method: "get",
+    });
+
+    const responseByIdMessage = await responseById.json();
+
+    // Then expect the response to return with our selected transformer
+    expect(responseById.status).toBe(200);
+    expect(responseByIdMessage.name).toBe(testTransformer);
+
+    // When the transformer is deleted
+    const responseAfterDelete = await fetch(
+      `${tansformerBaseUrl}/${toDeleteId}`,
+      {
+        method: "delete",
+      }
+    );
+
+    // Then expect a 204 resposne
+    expect(responseAfterDelete.status).toBe(204);
+  });
+
+  it("should not get the transformed message with transformed status code after transformer is deleted", async () => {
+    expect.assertions(2);
+
+    // When hello world is requested
+    const response = await fetch(baseUrl, {
+      method: "get",
+    });
+
+    // Then expect the reponse to not be transformed
+    const message = await response.json();
```

Comment:
Nitpick

This could be named better. Currently the object structure is `message : { message: string }`. When accessing the object it would `message.message` which breaks the whole philosophy of self documenting code

Reply:
Renamed it to `responseBody`